GameDefinition/Content/ItemKind.hs

```diff
@@ -1739,3 +1739,24 @@ currency = currencyTemplate
                ++ iaspects currencyTemplate
   , ieffects = [RefillCalm (-1)]
   }
+  
+-- * Clothing
+
+smokingJacket = ItemKind
+  { isymbol  = _symbolClothes
+  , iname    = "smoking jacket"
+  , ifreq    = [("common item", 100), ("misc clothing", 1)]
+  , iflavour = zipPlain [Green]
+  , icount   = 1
+  , irarity  = [(1, 9), (10, 3)]
+  , iverbHit = "stroke"
+  , iweight  = 5000
+  , idamage  = 0
+  , iaspects = [ Timeout $ (1 `d` 2) * 3
+               , AddSkill SkSpeed 5
```

Comment:
My initial idea was that it took up an armour slot, so you had to worry about defence more. 

How would you suggest limiting it?

Reply:
Oh, that would make perfects sense, but we don't have specialized slots (10 general equipment slots, so you can wear your jacket under ring mail and the either jackets on top). 

I'd think you could either limit the bonus to 2 (looking at other content, that's still quite good for the price of one item slot), add some drawback (we can brainstorm, an obvious one is `AddSkill SkOdor 2`, but it's not enough) or make it a unique artifact (quite powerful, but that's OK, it's a very game-defining item; the odour drawback is enough then).

Reply:
What about a bonus of two, plus the drawback?

Reply:
Works for me, though only playtesting will tell, so I may end up bumping it up, because the item is special, so it should not be "meh, I hope to find something better soon for that slot". :)

You could also go for a random value of speed, odour or both. And best make it scale with generation depth (e.g., see the `SkHurtMelee` value of darts). Then, rarely the item may be very good, adding to the excitement and also players may hoard the jackets (and a couple of other item kinds) to eventually use scrolls of transfiguration on them that re-roll the random item stats, giving a chance to get the perfect specimen (max speed, min odour). However, ensure proper granularity, so that only a few (4 to 8?) different kinds of jackets may ever be rolled so that a lucky hoarding player doesn't end up with inventory full of different jackets (UI nightmare), but instead with several copies of each of the few.

Reply:
I've set it to speed 2 and odor 2 for now, so the item can exist in a basic form while I work out how to do the more complex generation.

Reply:
Yep, now I see it's actually quite tricky, because speed is not very granular (perhaps speed 1 should mean 0.01m/s, not 0.1m/s as it is now, 2m/s == speed 20 being the normal human speed? what do you think?). To generate speed 4 only 1 in 100 times, we'd need (in the simplified version that takes no account of generation depth) something like `1 + max 0 (-97 + 10 \`d\` 10)`. I hope it works, because I think I've never used `max` for dice so far.

Anyway, if you'd like me to merge now, I'm perfectly fine with that. Or perhaps you'd like to tweak the smoking rooms as I outlined in some outdated code comment?

Reply:
Merging now works for me - I want to take a stab at potion descriptions next, if that's alright,, and then probably come back to the smoking rooms in the future.

Reply:
Splendid. It's a great item as is. Re potions, anything you find with `idesc    = ""`, commented out or not, is yours, because no items should have null description (though many may have a common one, just as is the case with potions so far now). If you fancy changing any that share a common description, optionally adding a label and/or tweaking it to match some new theme, be my guest. :)

Reply:
Regarding `1 + max 0 (-97 + 10 'd' 10)`, the `Ord` distance for `Dice` was not the right one, so I removed it and added a standalone `maxDice` operation, so `1 + maxDice 0 (-97 + 10 'd' 10)` should probably work fine here.